encrypted-media-respec.html

```diff
@@ -2101,6 +2101,30 @@
           <p>The following steps are run:</p>
           <ol>
             <li><p>Let the <var>session</var> be the associated <a>MediaKeySession</a> object.</p></li>
+            <li><p>Run the <a def-id="destroy-keys-and-licenses-algorithm"></a> algorithm.</p></li>
+            <li><p>Run the <a def-id="update-key-statuses-algorithm"></a> algorithm on the <var>session</var>, providing an empty sequence.</p></li>
+            <li><p>Run the <a def-id="update-expiration-algorithm"></a> algorithm on the <var>session</var>, providing <code>NaN</code>.</p></li>
+            <li><p>Let <var>promise</var> be the <a def-id="closed"></a> attribute of the <var>session</var>.</p></li>
+            <li><p>Resolve <var>promise</var>.</p></li>
+          </ol>
+        </section>
+        <section id="destroy-keys">
+          <h4>Destroy Keys and Licenses</h4>
+          <p>
+            The Destroy Keys and Licenses algorithm is run when the CDM needs to destroy all the keys and licenses associated with a session that
+            have not been explicitly stored.
+          </p>
+          <p>
+            For sessions of type <a def-id="temporary-session"></a> and <a def-id="persistent-usage-record-session"></a>, this algorithm MUST be
+            executed when the <a>MediaKeySession</a> object becomes inaccessible to the page, if the <a def-id="session-close-algorithm"></a>
```

Comment:
I'm not sure it makes sense to specify behavior for an object after it has become "inaccessible to the page." Note that this is a user agent algorithm (see the comment at line 2127 below) for an object that has become "inaccessible to the page." As far as the spec is concerned, the object no longer exists and cannot be executing algorithms.

Additionally, `persistent-usage-record` requires that this algorithm be run even after the page is destroyed, which is not addressed by this text and is not currently addressable by a user agent algorithm (per the discussion #85).